include/range/v3/view/indices.hpp

```diff
@@ -0,0 +1,123 @@
+/// \file
+// Range v3 library
+//
+//  Copyright Eric Niebler 2013-2014
+//  Copyright Gonzalo Brito Gadeschi
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/ericniebler/range-v3
+//
+
+#ifndef RANGES_V3_VIEW_INDICES_HPP
+#define RANGES_V3_VIEW_INDICES_HPP
+
+#include <meta/meta.hpp>
+#include <range/v3/detail/satisfy_boost_range.hpp>
+#include <range/v3/range_fwd.hpp>
+#include <range/v3/range_concepts.hpp>
+#include <range/v3/utility/static_const.hpp>
+#include <range/v3/view/take_exactly.hpp>
+#include <range/v3/view/iota.hpp>
+
+namespace ranges
+{
+    inline namespace v3
+    {
+
+        namespace view
+        {
+            /// Half-open range of indices: [from, to).
+            struct indices_fn
+              : iota_view<std::size_t>
```

Comment:
> That said, I'm a bit uncomfortable with drifting from unsigned to signed here and there, in dribs and drabs.

I have an eventual goal in mind. I've got a rough branch that makes `ranges::size` always return a signed type, and changes all of the `size_type`s of the views to be signed types. I've got the beginnings of a semantic wrapper `nonnegative<T>` that "knows" that the stored `T` value is not negative. What I haven't done yet is wrap all of that up nicely to see how it works together and interacts with the standard library.

I think your "Dribs and drabs" argument has sold me. Changing this component to use a signed type as well is not a significant burden, and I'd much rather present a holistic design than try to explain why a single component is trying to swim against the current.